tracing/src/main/java/com/palantir/tracing/Tracer.java

```diff
@@ -274,4 +286,17 @@ static void setTrace(Trace trace) {
         MDC.put(Tracers.TRACE_ID_KEY, trace.getTraceId());
     }
 
+    private static Trace getOrCreateCurrentTrace() {
+        Trace trace = currentTrace.get();
+        if (trace == null) {
+            trace = createTrace(Optional.empty(), Tracers.randomId());
+            setTrace(trace);
+        }
+        return trace;
+    }
+
+    private static void clearCurrentTrace() {
```

Comment:
Looks like the only two places this can be called are:

- `public Trace getAndClearTrace()`
- when a call to `fastCompleteSpan()` -> `popCurrentSpan()` (pops the last remaining span)

Reply:
That is correct